miner/miner.go

```diff
@@ -168,18 +131,37 @@ func (self *Miner) update() {
 	}
 }
 
-func (self *Miner) Start(coinbase common.Address) {
+func (self *Miner) Start(coinbase common.Address, e *rootchain.RootChainEpochPrepared, empty bool) {
 	atomic.StoreInt32(&self.shouldStart, 1)
 	self.SetEtherbase(coinbase)
 
 	if atomic.LoadInt32(&self.canStart) == 0 {
 		log.Info("Network syncing, will start miner afterwards")
 		return
 	}
-	if atomic.LoadInt32(&self.canStart) == 2 {
-		log.Info("Preparing epoch, will start miner afterwards")
+
+	if empty {
+		self.env = rawdb.ReadEpochEnv(self.db)
+		self.worker.start()
+		log.Info("current epoch is resumed")
 		return
 	}
+
+	self.env.SetIsRequest(e.IsRequest)
+	self.env.SetUserActivated(e.UserActivated)
+	self.env.SetRebase(e.Rebase)
+	self.env.SetCompleted(false)
+	self.env.SetNumBlockMined(big.NewInt(0))
```

Comment:
There is no `e.NumBlockMined` field. Because the parameter `e` is `*rootchain.RootChainEpochPrepared` type. So I just set `big.NewInt(0)` to reset the number of block mined.